src/main/java/codeu/controller/FileUploadServlet.java

```diff
@@ -0,0 +1,41 @@
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Part;
+
```

Comment:
Since we're using multipart form data in profile.jsp, we want the @MultipartConfig annotation here. I suggest @MultipartConfig(maxFileSize = 16177215 ) giving us a max file size of 16MB
I also suggest the @WebServlet("profile/FileUploadServlet") annotation since it will allow us to get the image from the doGet() of this servlet. Essentially, when we post the image with <img src="FileUploadServlet"> this will allow us to get to this servlet's doGet() function.

Reply:
As Fang noted, we need a URL mapping and using the above annotation gives this to us without having to change anything in the web.xml file

Reply:
Just added this part! Still working on your other comments